cmake/Dependencies.common.cmake

```diff
@@ -303,7 +304,7 @@ if(BUILD_NVIMAGECODEC)
     message(STATUS "NVIMGCODEC_DEFAULT_INSTALL_PATH=${NVIMGCODEC_DEFAULT_INSTALL_PATH}")
     add_definitions(-DNVIMGCODEC_DEFAULT_INSTALL_PATH=\"${NVIMGCODEC_DEFAULT_INSTALL_PATH}\")
 
-    set(DALI_INSTALL_REQUIRES_NVIMGCODEC "\'nvidia-nvimgcodec-cu${CUDA_VERSION_MAJOR} >= ${NVIMGCODEC_REQ_VERSION}',")
+    set(DALI_INSTALL_REQUIRES_NVIMGCODEC "\'nvidia-nvimgcodec-cu${CUDA_VERSION_MAJOR} >= ${NVIMGCODEC_REQ_MIN_VERSION}, <= ${NVIMGCODEC_REQ_MAX_VERSION}',")
```

Comment:
```suggestion
    set(DALI_INSTALL_REQUIRES_NVIMGCODEC "\'nvidia-nvimgcodec-cu${CUDA_VERSION_MAJOR} >= ${NVIMGCODEC_REQ_MIN_VERSION}, < ${NVIMGCODEC_REQ_MAX_VERSION}',")
```

Reply:
Like in https://github.com/NVIDIA/DALI/pull/5709/files

Reply:
I don't think that's right. In case of nvimagecodec we know a particular version which doesn't work.
In case of other libraries, we know which was the last known working version. How do we guess the next one?

Reply:
I agree; we know which version doesn't work. I doubt the version with the new API will appear as a patch for 0.3.

Reply:
Sure. I just cherry-picked @jantonguirao PR and added other bounds on top of it.